### PR TITLE
Add a test for branch ref name merge

### PR DESCRIPTION
**Pull request by kshyatt**
Another code path missed...

### CODE REVIEW

File: test/libgit2.jl [hunk 1]
```diff
@@ -705,6 +705,26 @@ mktempdir() do dir
...
+            oldhead = LibGit2.head_oid(repo)
+            LibGit2.branch!(repo, "branch/ff_c")
+            open(joinpath(LibGit2.path(repo),"ff_file5"),"w") do f
+                write(f, "555\n")
+            end
+            LibGit2.add!(repo, "ff_file5")
+            LibGit2.commit(repo, "add ff_file5")
+
+            open(joinpath(LibGit2.path(repo),"ff_file6"),"w") do f
+                write(f, "666\n")
```
Comment (ararslan) [inline test/libgit2.jl]: 🤘

**Review by ararslan — APPROVED** [review]
LGTM. Nice work as always.

**Comment by kshyatt** [comment]
Hopefully CI will keep running on the old commit? I'll get rid of `[ci skip]` if not. Anyways this caught me when I was writing this test so I thought the `!!! note` was apropos.

**Comment by ararslan** [comment]
Yes, good call on the doc note. Very nice addition.

**Comment by kshyatt** [comment]
CI passed and I'll edit out the `[ci skip]`.

**Pull request merged**